### ISSUE TITLE
Support ExposedPorts and Volumes in image config.

### ISSUE THREAD

**Issue by Random-Liu** [kubernetes-incubator/cri-containerd#186]
Currently, we are ignoring those fields.
We should:
1) Support Volumes based on the definition in oci image spec.
2) Consider whether we need to support ExposedPorts.

**Comment by Random-Liu** [comment]
The volumes definition in image spec is not very clear, but I assume it's expected behavior is similar with [Dockerfile VOLUME](https://docs.docker.com/engine/reference/builder/#volume).

Actually, we don't use docker volumes in Kubernetes. However, there may be user relying on image volumes (e.g. `mongo:latest` relies on volumes `/data/configdb` and `/data/db`). There 2 options for volumes:
1) Volume directory and content should be in image already, we could ignore it as what we are doing today. However, a problem with this is that if container rootfs is specified as read-only, the volume directories will be read-only, too. This will be a regression.
2) Do what docker does, create local file; copy volume content into the file, and mount the file into container as read-write. There is no regression for this approach, but we need to add helper in containerd to do this, it will be similar with `WithUserID` or `WithUsername` which require pre-access to container root filesystem.

@ijc @crosbymichael Any opinions?

**Comment by ijc** [comment]
> Actually, we don't use docker volumes in Kubernetes

I think it is more accurate to say you don't explicitly use them, it is unfortunately the case that they are implictly relied on though.

Sadly I don't think your no.1 is a viable option, the one [demo](https://github.com/microservices-demo/microservices-demo) I've tried so far relies on them and I expect it to be quite widespread.

So no.2 seems like the only reasonable short term option. Perhaps longer term Kubernetes itself could spot these things and create explicit kube volumes to cover them? AIUI that would be good for accounting etc anyway (I see you filed that as an issue while I wrote the following paragraph!)

On that note the behaviour on no.2 needs to only happen if there is no user bind to the same path. The Docker logs is in [oci_linux.go](https://github.com/moby/moby/blob/master/daemon/oci_linux.go#L498..L514)
